package/MDAnalysis/analysis/pca.py

```diff
@@ -414,6 +414,129 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
+
+            pca = PCA(universe, select='backbone').run()
+
+        Obtain a transformation function to project the
+        backbone trajectory onto the first principal component::
+
+            project = pca.project_single_frame(components=0)
+
+        To projet onto the first two components, run::
+
+            project = pca.project_single_frame(components=[0,1])
+
+        Alternatively, the transformation can be applied to PCA atoms and
+        extrapolated to other atoms according to the CA atom's translation
+        in each residue::
+
+            all = u.select_atoms('all')
+            project = pca.project_single_frame(components=0,
+                                               group=all, anchor='name CA')
+
+        Finally, apply the transformation function to a timestep::
+
+            project(u.trajectory.ts)
+
+        or apply the projection to the universe::
+
+            u.trajectory.add_transformations(project)
+
+
+        .. versionadded:: 2.2.0
+        """
+        if not self._calculated:
+            raise ValueError('Call run() on the PCA before projecting')
+
+        if group is not None:
+            if anchor is None:
+                raise ValueError("'anchor' cannot be 'None'" +
+                                 " if 'group' is not 'None'")
+            anchors = group.select_atoms(anchor)
+            anchors_res_ids = anchors.resindices
+            if np.unique(anchors_res_ids).size != anchors_res_ids.size:
+                raise ValueError("More than one 'anchor' found in residues")
+            if not np.isin(group.resindices, anchors_res_ids).all():
+                raise ValueError("Some residues in 'group'" +
+                                 " do not have an 'anchor'")
+            if not anchors.issubset(self._atoms):
+                raise ValueError("Some 'anchors' are not part of PCA class")
+
+            non_pca = group - self._atoms
+            pca_res_indices, pca_res_counts = np.unique(
+                self._atoms.resindices, return_counts=True)
+
+            matrix_extrapolate = np.zeros((non_pca.n_atoms, anchors.n_atoms))
+            cumul_atoms_old = 0
+            for res in group.residues:
+                res_index = res.resindex
+                n_common = pca_res_counts[np.where(
+                           pca_res_indices == res_index)][0]
+                cumul_atoms_new = (cumul_atoms_old +
+                                   res.atoms.n_atoms - n_common)
+                matrix_extrapolate[cumul_atoms_old:cumul_atoms_new,
+                                   res_index] = 1
+                cumul_atoms_old = cumul_atoms_new
+
+        if components is None:
+            components = range(self.results.p_components.shape[1])
+
+        def wrapped(ts):
+
+            if group is not None:
+                anchors_pos = np.matmul(matrix_extrapolate, anchors.positions)
+                non_pca.positions -= anchors_pos
```

Comment:
added the `@` operator.

> ... keep `anchors_pos` because you can reuse them for the backtransformation below! (At least if I understood the code correctly.)

`anchors` are (required to be) part of `self` atomgroup, which is projected with the inverse-PCA in the next line. So, need to recalculate later.

Reply:
Thanks for the explanation, I missed the fact that anchors are also being transformed as part of atomgroup.